### PR TITLE
[PKG-6472] Initial build 1.12.0

### PR DESCRIPTION
**Pull request by ViridianMelody**
dbt-common 1.12.0 ❄️

**Destination channel:** Snowflake

### Links

- [PKG-6472] 
- [Upstream repository](https://github.com/dbt-labs/dbt-common)

### Explanation of changes:

- Tests aren't included in Pypi source and there aren't any tags or releases on Github.

[PKG-6472]: https://anaconda.atlassian.net/browse/PKG-6472?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -29,25 +30,31 @@ requirements:
...
 test:
+# Tests not included in pypi source, and github has no tagged releases.
   imports:
     - dbt_common
   commands:
     - pip check
   requires:
     - pip
-    - python {{ python_min }}
+    - python
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]:
```suggestion
```
Here it's redundant as it's already in run

File: recipe/meta.yaml [hunk 2]
```diff
@@ -29,25 +30,31 @@ requirements:
     - jinja2 >=3.1.3,<4
     - mashumaro >=3.9,<4.0
     - pathspec >=0.9,<0.13
-    - protobuf >=5.0,<6.0
+    - protobuf >=4.0.0,<5.0.0
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:33]: Should we increase the upper bound to <6.0? https://github.com/dbt-labs/dbt-common/blob/main/pyproject.toml#L38

Comment (ifitchet) [reply recipe/meta.yaml:33]: Isn't `protobuf` one of those where the pinnings are quite important for interop?  I would have thought we keep it to  within the one major version, ie. 4.* or 5.*.  Do we have any `protobuf` guidelines written down for reference?

**Review by ifitchet — APPROVED** [review]

File: recipe/meta.yaml (hunk 2 above)
Comment (ViridianMelody) [reply recipe/meta.yaml:33]: There's no tag for this release on Github, but the pypi source has `"protobuf>=4.0.0,<5.0.0",`

**Pull request merged**